application/repository/exister.go

```diff
@@ -17,17 +17,26 @@ type Exister interface {
 	CheckExists(ctx context.Context, id string) error
 }
 
-// Exists returns true if an item exists in the database table with a given ID
-func Exists(ctx context.Context, db *gorm.DB, tableName string, id string) (bool, error) {
+// Exists returns true if a soft or hard deletable item exists in the database table with a given ID
+func Exists(ctx context.Context, db *gorm.DB, tableName string, id string, softDeletable bool) (bool, error) {
```

Comment:
Could we something like `isSoftDeleted` ?
That would imply "check if object exists given that the object is soft deleted"